src/plugins/graph/components/graph.tsx

```diff
@@ -129,10 +130,10 @@ export const Graph = observer(
     graphModel.config.setAttributeType(graphPlaceToAttrRole[place], treatAs);
     dataset && graphController?.handleAttributeAssignment(place, dataset.id, attrId);
 
-    const connectingLines = graphModel.adornments.find(a => a.type === "Connecting Lines");
+    const connectingLines = graphModel.adornments.find(a => a.type === kConnectingLinesType);
     if (connectingLines && place === "left") {
-      treatAs === 'categorical' && graphModel.hideAdornment("Connecting Lines");
-      treatAs === 'numeric' && graphModel.showAdornment(connectingLines, "Connecting Lines");
+      treatAs === 'categorical' && graphModel.hideAdornment(kConnectingLinesType);
+      treatAs === 'numeric' && graphModel.showAdornment(connectingLines);
```

Comment:
As a future improvement, would it make sense to move this hide/show code out of `handleTreatAttrAs` and instead have the adornment code itself decide whether to display the lines based on the current type of axes? Since there are other ways that axes can change type.